streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -273,15 +272,27 @@ public StreamsConfig(Map<?, ?> props) {
         props.put(ProducerConfig.LINGER_MS_CONFIG, "100");
 
         // remove properties that are not required for producers
+        removeStreamsSpecificConfigs(props);
         props.remove(StreamsConfig.KEY_DESERIALIZER_CLASS_CONFIG);
         props.remove(StreamsConfig.VALUE_DESERIALIZER_CLASS_CONFIG);
-        props.remove(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG);
+        props.remove(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG);
```

Comment:
Do we want to always enforce "latest" for AUTO_OFFSET_RESET_CONFIG? I think there are some cases where a user may want to consume from earliest.